src/metrics/prometheus.js

```diff
@@ -0,0 +1,88 @@
+// @flow
+// Copyright (c) 2018 Jaeger Author.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+
+class CounterPromWrapper {
+  _counter: any;
+
+  constructor(counter: any) {
+    this._counter = counter;
+  }
+
+  increment(delta: number): void {
+    this._counter.inc(delta);
+  }
+};
+
+class GaugePromWrapper {
+  _gauge: any;
+
+  constructor(gauge: any) {
+    this._gauge = gauge;
+  }
+
+  update(value: number): void {
+    this._gauge.set(value);
+  }
+};
+
+export default class PrometheusMetricsFactory {
+  _cache: any = {};
+  _namespace: ?string;
+  _promClient: any;
+
+  constructor(promClient: {}, namespace: ?string) {
```

Comment:
it would be good to have method docs showing how this should be instantiated (similar to what's added to the README).